pkg/apis/core/v1/resource_tags.go

```diff
@@ -0,0 +1,74 @@
+package v1
+
+import (
+	"context"
+	"fmt"
+
+	"go.anx.io/go-anxcloud/pkg/api"
+	"go.anx.io/go-anxcloud/pkg/api/types"
+)
+
+// Tag adds tags to an object resource
+func Tag(ctx context.Context, a api.API, obj types.IdentifiedObject, tags ...string) error {
+	objects, err := resourceWithTagObjects(obj, tags...)
+	if err != nil {
+		return fmt.Errorf("failed to tag resource: %w", err)
+	}
+
+	for _, obj := range objects {
+		if err := a.Create(ctx, obj); err != nil {
+			if err, ok := err.(api.HTTPError); ok && err.StatusCode() == 422 {
+				// already tagged -> skip
+				continue
+			}
+			return fmt.Errorf("failed to tag resource: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// Untag removes tags from an object resource
+func Untag(ctx context.Context, a api.API, obj types.IdentifiedObject, tags ...string) error {
+	objects, err := resourceWithTagObjects(obj, tags...)
+	if err != nil {
+		return fmt.Errorf("failed to untag resource: %w", err)
+	}
+
+	for _, obj := range objects {
+		if err := a.Destroy(ctx, obj); api.IgnoreNotFound(err) != nil {
+			return fmt.Errorf("failed to untag resource: %w", err)
+		}
+	}
+	return nil
+}
+
+func resourceWithTagObjects(obj types.IdentifiedObject, tags ...string) ([]*ResourceWithTag, error) {
+	identifier, err := api.GetObjectIdentifier(obj, true)
+	if err != nil {
+		return nil, err
+	}
+
+	objects := make([]*ResourceWithTag, 0, len(tags))
+
+	for _, tag := range tags {
+		objects = append(objects, &ResourceWithTag{Identifier: identifier, Tag: tag})
+	}
+
+	return objects, nil
+}
+
+// ListTags retrieves tags of given object
+func ListTags(ctx context.Context, a api.API, obj types.IdentifiedObject) ([]string, error) {
+	identifier, err := api.GetObjectIdentifier(obj, true)
+	if err != nil {
+		return nil, err
+	}
+
+	r := &Resource{Identifier: identifier}
+	if err := a.Get(ctx, r); err != nil {
+		return nil, fmt.Errorf("failed to fetch resources tags: %w", err)
```

Comment:
If you decide in `Tag` and `Untag` to return the second error as-is, please do here, too